src/effective-dart/usage.md

```diff
@@ -404,10 +410,15 @@ class UploadException {
 
 Be careful when using a local variable. If you need to write back to the field,
 make sure that you don't write back to the local variable instead. (Making the
-local variable `final` can prevent such mistakes.) Also, if the field might
+local variable [`final`] can prevent such mistakes.) Also, if the field might
 change while the local is still in scope, then the local might have a stale
-value. Sometimes it's best to simply use `!` on the field.
+value. Sometimes it's best to simply [use `!`] on the field.
 
+[Fixing type promotion failures]: /tools/non-promotion-reasons
+[private]: design#prefer-making-declarations-private
+[final]: design#prefer-making-fields-and-top-level-variables-final
```

Comment:
I guess we haven't really documented this anywhere yet (I will), but try to avoid relative links:

```suggestion
[private]: /effective-dart/design#prefer-making-declarations-private
[final]: /effective-dart/design#prefer-making-fields-and-top-level-variables-final
```